### PR TITLE
Adding product function for sorted set instances

### PR DESCRIPTION
**Pull request by calvinbrown085**
https://github.com/typelevel/cats/issues/2708
I ran `sbt prePR`, i'm kind of confused why it's failing the bin compatibility check if anyone has a second to look :)

### CODE REVIEW

File: core/src/main/scala/cats/instances/all.scala [hunk 1]
```diff
@@ -42,6 +42,7 @@ trait AllInstancesBinCompat1
     with StreamInstancesBinCompat0
     with MapInstancesBinCompat0
     with SortedMapInstancesBinCompat0
+    with SortedSetInstancesBinCompat0
```
Comment (LukaJCB) [inline core/src/main/scala/cats/instances/all.scala]: This is the problem right here, you can't add anything to existing traits, only final classes and objects :)

Comment (calvinbrown085) [reply core/src/main/scala/cats/instances/all.scala]: Ah, Do I just need to add `AllInstancesBinCompat4` that extends `SortedSetInstancesBinCompat0` ?

Comment (LukaJCB) [reply core/src/main/scala/cats/instances/all.scala]: Yup,I believe so

**Comment by LukaJCB** [comment]
Can you also mix it into `cats.instances.all` and `cats.implicits`?

**Comment by calvinbrown085** [comment]
@LukaJCB Yep!

File: tests/src/test/scala/cats/tests/SortedSetSuite.scala [hunk 2]
```diff
@@ -2,13 +2,17 @@ package cats
...
+import cats.laws.discipline.{FoldableTests, SemigroupKTests, SemigroupalTests, SerializableTests}
 import cats.kernel.laws.discipline.{BoundedSemilatticeTests, HashTests, PartialOrderTests}
 import cats.laws.discipline.arbitrary._
 
 import scala.collection.immutable.SortedSet
 
 class SortedSetSuite extends CatsSuite {
+
+  checkAll("SortedSet[Int]", SemigroupalTests[SortedSet].semigroupal[Int, Int, Int])
+  checkAll("Semigroupal[SortedSet]", SerializableTests.serializable(Semigroupal[SortedSet]))
```
Comment (LukaJCB) [inline tests/src/test/scala/cats/tests/SortedSetSuite.scala]: `Semigroupal[SortedSet]` isn't in scope here :)

Comment (LukaJCB) [reply tests/src/test/scala/cats/tests/SortedSetSuite.scala]: You can mix it into `CatsSuite`

**Pull request closed without merging**